### DISCUSSION TITLE
No MSG5 detected

### DISCUSSION THREAD

**Discussion by mwasserstein** [ARM-DOE/pyart#1233] [Q&A] [answered]
When using `pyart.io.read` for several radar files, I occasionally get the error message `No MSG5 detected. Setting to meaningless data. Rethink your life choices and be ready for errors.Specifically fixed angle data will be missing`. I'm wondering what exactly this message means. Could this mean that I downloaded the data incorrectly? Or is it a faulty file with data that I cannot use? And would there be any way to get around this error? Thanks for your help!

**Comment by jrobrien91** [comment]
Hi Michael, do you have an example radar file that results in this error?

**Reply by mwasserstein** [reply]
Here is one radar file that will cause the error.

[KMTX20000114_052707.gz](https://github.com/ARM-DOE/pyart/files/9292752/KMTX20000114_052707.gz)

**Comment by mgrover1** [comment, marked as answer]
That error comes from the NEXRAD Level 2 file reader. It means that the Volume Control Pattern (VCP) data is missing, and Py-ART cannot determine with elevation levels are present

https://github.com/ARM-DOE/pyart/blob/f923551e1fd8de72c8db6c76e6bc9913bf635644/pyart/io/nexrad_level2.py#L168#L171

You can read in the file, but it is missing important metadata which may limit your analysis.

**Reply by mwasserstein** [reply]
Ok. Any idea what may be causing the file to not have VCP data? I download the data from AWS.

**Reply by mgrover1** [reply]
I am not sure - is it most of the files?

**Reply by mgrover1** [reply]
I would recommend reaching out to NOAA about this data quality issue. Py-ART reads in what it can (with a warning that it can't obtain the metadata since it's not there). You should be able to still work with the files, but be cautioned that you will not information about the elevation scans.

**Reply by mgrover1** [reply]
Here is the email for NCEI [ncei.info@noaa.gov](mailto:ncei.info@noaa.gov)

**Reply by mwasserstein** [reply]
I would say it is about 15% of the files I'm working with. I will need to keep investigating this. Thanks for your help!

**Reply by dopplershift** [reply]
Old data (not sure when it started) don't contain those messages. IMO, while it's a very convenient that those VCP messages are available, a robust reader needs to handle the case when it doesn't get that information, since the files are very much not corrupt or missing anything due to transmission.
